arm/Microsoft.DataFactory/factories/deploy.bicep

```diff
@@ -125,26 +134,21 @@ resource dataFactory 'Microsoft.DataFactory/factories@2018-06-01' = {
   }
 }
 
-resource dataFactory_managedVirtualNetwork 'Microsoft.DataFactory/factories/managedVirtualNetworks@2018-06-01' = {
-  parent: dataFactory
-  name: 'default'
-  properties: {}
+module dataFactory_managedVirtualNetwork 'managedVirtualNetwork/deploy.bicep' = {
+  name: '${uniqueString(deployment().name, location)}-ManagedVirtualNetwork'
+  params: {
+    name: managedVirtualNetworkName
+    dataFactoryName: dataFactory.name
+  }
 }
 
-resource dataFactory_integrationRuntime 'Microsoft.DataFactory/factories/integrationRuntimes@2018-06-01' = {
-  parent: dataFactory
-  name: 'AutoResolveIntegrationRuntime'
-  properties: {
-    type: 'Managed'
-    managedVirtualNetwork: {
-      referenceName: 'default'
-      type: 'ManagedVirtualNetworkReference'
-    }
-    typeProperties: {
-      computeProperties: {
-        location: 'AutoResolve'
-      }
-    }
+module dataFactory_integrationRuntime 'integrationRuntime/deploy.bicep' = {
+  name: '${uniqueString(deployment().name, location)}-IntegrationRuntime'
+  params: {
+    name: integrationRuntimeName
+    dataFactoryName: dataFactory.name
+    managedVirtualNetworkName: managedVirtualNetworkName
+    typeProperties: typeProperties
```

Comment:
Please move all integrationRuntime properties into a parameter `param integrationRuntime object`. We should try to keep child resource properties away from the parent.
Moving it to an object allows us to update the child without updating the parent parameters again.
After updating please re-run the readme generation script for the parent readme

Reply:
This has been updated. There is now an integrationRuntime object and have added the capability of deploying a managed or self-hosted Integration Runtime